src/main/java/Duke.java

```diff
@@ -1,10 +1,115 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
+        printHeader();
+
+        String input = "";
+        int i = 0;
+        int taskCount = 0;
+        Task[] tasks = new Task[100];
+
+        do {
+            input = getInput();
+
+            if (!input.equals("bye")) {
+                String[] splittedInput = input.split(" ");
+                if (input.equals("list")) {
+                    printListOutput(i, tasks);
+                } else if (splittedInput[0].equals("done")) {
+                    markTaskAsDone(tasks, splittedInput[1]);
+                } else {
+                    taskCount++;
+                    addNewTask(input, i, taskCount, tasks);
+                    i++;
+                }
+            }
+        } while (!input.equals("bye"));
```

Comment:
The main method is perhaps getting a bit too long. Perhaps it would be good to abstract this part out into another function?